data/mods/MindOverMatter/powers/learning_eocs/electrokinesis.json

```
@@ -256,6 +256,94 @@
       }
     ]
   },
+  {
+    "type": "effect_on_condition",
+    "id": "EOC_ELECTROKIN_LEARNING_ROBOT_INTERFACE",
+    "recurrence": [ "96 hours", "168 hours" ],
+    "condition": {
+      "and": [
+        { "u_has_trait": "ELECTROKINETIC" },
+        { "math": [ "u_vitamin('vitamin_psi_learning_counter')", "==", "1" ] },
+        {
+          "or": [
+            { "x_in_y_chance": { "x": 1, "y": 80 } },
+            {
+              "and": [
+                { "math": [ "u_spell_level('electrokinetic_see_electric')", ">=", "12" ] },
+                { "math": [ "u_spell_level('electrokinetic_hacking_interface')", ">=", "8" ] },
+                { "math": [ "u_spell_level('electrokinetic_kill_robot')", ">=", "8" ] }
+              ]
+            }
+          ]
+        },
+        {
+          "not": {
+            "u_has_any_effect": [
```

```suggestion
            "u_has_any_effect": [
```
Ideally I'd like to see this replaced with a flag check and a mainline effects flag "DISTRACTING" for any effect that is DISTRACTING to the PC and would interrupt complex processes.  @Standing-Storm   though seeing as how several of these might not be distracting such as pre_cold and pre_flu or being covered in spores might or might not be distracting?  Sorry am mostly thinking about ways to cut down on json bloat and make it something that other mods can contribute their effects to.

---

I would definitely love if there were some easy way to store all those conditions in one place and just reference them elsewhere so I didn't have 1000 lines of repeating them a hundred times. As you say, though, some things are distracting and some things are more just not conducive to having an insight (like the depression from eating protein bars)

---

I'll merge this for now and think on it.